src/common/allocator.h

```diff
@@ -0,0 +1,34 @@
+#ifndef ILTORB_ALLOCATOR_H
+#define ILTORB_ALLOCATOR_H
+
+#include <stddef.h>
+#include <stdint.h>
+
+struct Allocator {
+  Allocator() : allocated_unreported_memory(0) {}
+
+  int64_t allocated_unreported_memory;
+
+  struct AllocatedBuffer {
+    size_t size;
+    size_t available;
+    /* char data[...]; */
```

Comment:
Yes, the actual content starts right behind the “official” members of `AllocatedBuffer`.